### PR TITLE
feat: validate all .tf files

### PR DESCRIPTION
**Pull request by kassianh**
https://app.shortcut.com/mineiros/story/5021/terradoc-validate-should-validate-against-all-tf-files-in-dir

### CODE REVIEW

**Review by mariux — CHANGES_REQUESTED** [review]
hi @kassianh looks good, but it fails if terraform includes blocks like `data "xxx" "yyy" {}` and other terraform blocks... such blocks and others should just be ignored as we only care about variables and outputs..

File: cmd/terradoc/cli/validate.go [hunk 1]
```diff
@@ -80,17 +108,40 @@ func (vcm ValidateCmd) Run() error {
 	return nil
 }
 
-func printValidationSummary(summary validators.Summary, docFilename, defFilename string) {
+func printValidationSummary(summary validators.Summary, docFilename string) {
 	for _, missingDef := range summary.MissingDefinition {
-		fmt.Fprintf(os.Stderr, "Missing %s definition: %q is not defined in %q\n", summary.Type, missingDef, defFilename)
+		fmt.Fprintf(os.Stderr, "Missing %s definition: %q is not defined in .tf files\n", summary.Type, missingDef)
```
Comment (mariux) [inline cmd/terradoc/cli/validate.go:113]: this is the other way around: it is NOT a missing definition but a documentation for a non existing variable.

```suggestion
		fmt.Fprintf(os.Stderr, "Unknown %s documented: %q is not defined in any .tf files\n", summary.Type, missingDef)
```

File: cmd/terradoc/cli/validate.go [hunk 2]
```diff
@@ -80,17 +108,40 @@ func (vcm ValidateCmd) Run() error {
...
 	}
 
 	for _, missingDoc := range summary.MissingDocumentation {
 		fmt.Fprintf(os.Stderr, "Missing %s documentation: %q is not documented in %q\n", summary.Type, missingDoc, docFilename)
```
Comment (mariux) [inline cmd/terradoc/cli/validate.go:117]: docFilename should not be the absolute path but the name of the parameter given.

**Review by mariux — DISMISSED** [review]
please improve texts in reporting of validation

File: cmd/terradoc/cli/validate.go (hunk 2 above)
Comment (kassianh) [reply cmd/terradoc/cli/validate.go:117]: @mariux Could you clarify this? What do you mean by the name of the parameter given?

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**